pkg/kubelet/allocation/allocation_manager.go

```diff
@@ -61,18 +64,56 @@ type Manager interface {
 
 	// RemoveOrphanedPods removes the stored state for any pods not included in the set of remaining pods.
 	RemoveOrphanedPods(remainingPods sets.Set[types.UID])
+
+	// Run starts the allocation manager. This is currently only used to handle periodic retry of
+	// pending resizes.
+	Run()
+
+	// PushPendingResize queues a pod with a pending resize request for later reevaluation.
+	// If a change in the resize queue is detected, pending resizes will be retried.
+	PushPendingResize(pod *v1.Pod)
+
+	// RetryPendingResizes signals the allocation manager to retry all pending resize requests.
+	RetryPendingResizes()
+
+	// HandlePodResourcesResize determines if a pod resize is feasible. If so,
+	// it updates the allocation and sets the pod resize conditions accordingly.
+	HandlePodResourcesResize(pod *v1.Pod) error
```

Comment:
Does this need to be exported through the interface? It looks like all the callers are internal?